### ISSUE TITLE
Developer Finder Tab

### ISSUE THREAD

**Issue by davidruvolo51** [davidruvolo51/travel-app#3]
This page will have the following elements

- [x] user preferences form
- [x] top three recommendations displayed as a map
- [x] add form option for country filter
- [x] add form option to remove top n cities
- [x] ~~summary metrics on all places (the entire dataset)~~
- [x] ~~summary of why the three recommendations were returned~~
- [x] ~~summary of the three recommendations~~

**Comment by davidruvolo51** [comment]
The summary of all places was moved to the data page

**Comment by davidruvolo51** [comment]
- [x] The user preferences formula needs to be fixed adjust the handling of the weights and ratings

**Comment by davidruvolo51** [comment]
Additional UI components were added for the user preferences form. This includes an option to limit the results to specific counties or country, as well as removing the number of top cities from the results.

In terms of the user preferences function, the formula is now correct and working (ratings was incorrectly used). The issues that plagued the application before was that the weights were being submitted in the incorrect order (I forgot that I set order of the weights to alphabetical order to avoid confusion when linking the client and server. Good thing I remembered!). In addition, the formula incorrectly calculated the weighted mean. I had divided the score by the sum of scores rather than the sum of weights.

There is still some work that needs to be done. The reference dataset needs to account for limits and country filters. The top cities filter also needs to be linked with the countries selected. I think this is a simple two step process.

```r
reference <- reactive({
      refs_filtered <- refs
      if (input$countryFilter != "") {
         refs_filtered <- refs %>% filter(country %in% input$countryFilter)
      } 

     if (input$limitsFilter != 0) {
         refs_filtered <- refs_filtered %>% 
                   arrange(-score) %>%
                   slice(seq_len(input$limitsFIlter))
     } 
      return(refs_filtered)
})
```

**Comment by davidruvolo51** [comment]
Additional UI components were added for the user preferences form. This includes an option to limit the results to specific counties or country, as well as removing the number of top cities from the results.

**Comment by davidruvolo51** [comment]
The limits filter and country filter were fixed in #6

**Comment by davidruvolo51** [comment]
I removed the last three tasks from the initial comment. I think it would be more useful to present the following visualisations

- [x] Some summary text that describes why the cities were selected
- [x] Vis on the number of place types for each recommended city (column chart)

**Comment by davidruvolo51** [comment]
Addressed in #7
